CHANGES

```diff
@@ -1,6 +1,111 @@
+  for more information.
+
+- We have adopted GitHub discussions
+  (https://github.com/UIUC-PPL/charm/discussions) as the preferred
+  venue for Charm++ questions and discussions instead of the
+  charm@cs.illinois.edu mailing list.
+
+- The BigSim emulation facility has been removed from the runtime
+  system.
+
+Charm++ Features & Fixes:
+
+- A significant overhaul of the load balancing infrastructure and the
+  addition of TreeLB, a new, more flexible and extensible base LB,
+  intended to replace the previous CentralLB and HybridBaseLB designs
+  (load balancers using these previous designs are still
+  supported). Some included strategies have been rewritten to use
+  TreeLB, unused and extraneous strategies have been removed. Please
+  see
+  https://charm.readthedocs.io/en/latest/charm++/manual.html#load-balancing
+  for more information.
+
+- Support for intra-node GPU messages, allowing data to be sent
+  to/from GPU memory without touching host memory.
+
+- Support for variable sized messages in TRAM.
+
+- Added a pup_buffer API with zero copy functionality.
+
+- Array broadcasts are now node aware, avoiding unnecessary
+  duplication of messages. Expedited nokeep array broadcasts are also
+  allowed now.
+
+- Added CmiNodeReduce API for node level reductions.
+
+- Builds with CMK_OPTIMIZE=1 will only segfault to aid in debugging in
+  CmiAbort when ++debug or +truecrash are provided.
+
+- Callbacks used in liveViz have been made ASLR safe.
+
+- Updated implementation of atomics, locks, and fences to use C++11/C11
+  versions where suitable.
+
+- Fixed bugs in HAPI and updated implementation to use new CUDA APIs.
+
+- Added whenidle attribute to indicate entry method to be called when
+  a PE is idle.
+
+- Improved performance, support, and fixes for UCX. IBM Power is now
+  supported with UCX.
+
+- The CmiSyncSend family now tries to avoid copies for nokeep
+  messages.
+
+- Fixed bug with element IDs with CMK_GLOBAL_LOCATION_UPDATE.
+
+- Fixed bug in BlockMap in array creation.
+
+- Added CcdPROCESSOR_LONG_IDLE to run a function during long periods
+  of idleness and CcdSCHEDLOOP to run a function during every
+  scheduler loop.
+
+- Added isCheckpoint() and isMigration() methods for users to
+  condition logic inside PUP based on its purpose.
+
+- Added execution metadata to Projections logs.
+
+- Added several new benchmarks and tests.
+
+- Bounded number of hops messages can take when chares frequently
+  migrate.
+
+Adaptive MPI:
+
+- Isomalloc sync is now enabled by default and the implementation no
+  longer uses the filesystem to pass data. Isomalloc sync can be
+  disabled with +no_isomalloc_sync.
+
+- Added AMPI-only build target, AMPI-only.
+
+- Fixed tlsglobals migration on macOS.
+
+- Fixed bugs in activation of migration callbacks.
+
+- Fixed a bug in MPI_Waitsome.
+
+- Accept arguments from AMPI_BUILD_FLAGS environment variable for
+  ampiCC.
+
+- Added PUP methods for derived datatypes.
```

Comment:
This doesn't need to be in the release notes, it's a minor implementation detail